utils/helpers/schedule/setDateToPeriodTime.ts

```diff
@@ -18,9 +18,10 @@ export default function setDateToPeriodTime(
   edge?: "start" | "end",
 ) {
   return new Date(
-    date.setHours(
-      PERIOD_TIMES[periodNumber + (edge === "end" ? 0 : -1)].hours,
-      PERIOD_TIMES[periodNumber + (edge === "end" ? 0 : -1)].min,
+    new Date(date).setHours(
```

Comment:
Fixing `periodNumberAt` returning an incorrect value seems to be quite troublesome and hard to find. Congratulations on that.